### PR TITLE
HDDS-4542. Need throw exception to trigger FailoverProxyProvider of SCM client to work

### PR DESCRIPTION
**Pull request by GlenGeng**
## What changes were proposed in this pull request?

ScmBlockLocationProtocolServerSideTranslatorPB and StorageContainerLocationProtocolServerSideTranslatorPB need throw exception to trigger FailoverProxyProvider of SCM client to work.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4542

## How was this patch tested?

CI

### CODE REVIEW

**Comment by GlenGeng** [comment]
@ChenSammi @timmylicheng @nandakumar131  Please take a look at this PR. This is one of the hot fix found during the phase 1.5 test.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/protocol/StorageContainerLocationProtocolServerSideTranslatorPB.java [hunk 1]
```diff
@@ -133,9 +133,7 @@ private boolean isLeader() throws ServiceException {
   public ScmContainerLocationResponse submitRequest(RpcController controller,
       ScmContainerLocationRequest request) throws ServiceException {
     if (!isLeader()) {
-      return ScmContainerLocationResponse.newBuilder()
-              .setCmdType(request.getCmdType()).setTraceID(request.getTraceID())
-              .setSuccess(false).setStatus(Status.SCM_NOT_LEADER).build();
+      throw new ServiceException(new IOException("SCM IS NOT LEADER"));
```
Comment (timmylicheng) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/protocol/StorageContainerLocationProtocolServerSideTranslatorPB.java:136]: Shall we expose scm names or ipadress for the current SCM?

**Comment by timmylicheng** [comment]
Thanks @GlenGeng 's contribution. Overall LGTM. Just left a comment inline.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/protocol/StorageContainerLocationProtocolServerSideTranslatorPB.java (hunk 1 above)
Comment (GlenGeng) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/protocol/StorageContainerLocationProtocolServerSideTranslatorPB.java:136]: @timmylicheng  Thanks for looking at this patch !

Good point !
As far as I know, the failover proxy provider will print out the IP address, along with the exception.

We need revisit this code in future, when the failover proxy provider should support leaderHint. I've file a jira to track this task, https://issues.apache.org/jira/browse/HDDS-4545

**Comment by ChenSammi** [comment]
+1

**Pull request merged**